tests/psbt.rs

```diff
@@ -0,0 +1,704 @@
+//! This file implements the test vector workflow in:
+//!
+//!  https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#test-vectors
+//!
+//! starting at the line:
+//!
+//!  The private keys in the tests below are derived from the following master private key:
+//!
+
+// Attribution: `sign_psbt` logic is based on code in the `wallet` module of the `bdk` project. The
+// code in the two functions `legacy_sighash()` and `segwit_v0_sighash` is copied directly from
+// `bdk`, hence their licence applies to that code.
+//
+// ref: http://github.com/bitcoindevkit/bdk
+//
+// Bitcoin Dev Kit
+// Written in 2020 by Alekos Filini <alekos.filini@gmail.com>
+//
+// Copyright (c) 2020-2021 Bitcoin Dev Kit Developers
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+extern crate bitcoin;
+
+use std::fmt;
+use std::str::FromStr;
+use std::collections::BTreeMap;
+
+use bitcoin::{Amount, Denomination, EcdsaSig, EcdsaSighashType, Network, OutPoint, PublicKey, PrivateKey, Script, Sighash, Transaction, Txid, TxOut, TxIn, Witness};
+use bitcoin::util::address::WitnessVersion;
+use bitcoin::consensus::encode::{serialize_hex, deserialize};
+use bitcoin::blockdata::{opcodes, script};
+use opcodes::all::OP_PUSHBYTES_0 as OP_0;
+use bitcoin::hashes::Hash;
+use bitcoin::hashes::hex::{FromHex};
+use bitcoin::secp256k1::{self, Message, SECP256K1};
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey, Fingerprint, IntoDerivationPath, KeySource};
+use bitcoin::util::psbt::{PartiallySignedTransaction, PsbtSighashType};
+use bitcoin::util::sighash::SighashCache;
+
+const NETWORK: Network = Network::Testnet;
+
+type Psbt = PartiallySignedTransaction; // TODO: Consider adding this alias to `rust-bitcoin`.
```

Comment:
I think we should. I use this shorthand for `rust-miniscript`

Reply:
Done in #951, thanks.